src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,140 @@
+            ui.showNewTask(event);
+        } else if (lowerCase.startsWith("todo")) {
+            try {
+                String item = input.substring(5);
+                Task todo = new Todo(item);
+                taskList.add(todo);
+                storage.add(todo);
+                ui.showNewTask(todo);
+            } catch(StringIndexOutOfBoundsException e) {
+                throw new DukeException("OOPS!!! The description of a event cannot be empty.");
+            }
+        } else if (lowerCase.startsWith("done")) {
+            try {
+                char temp = input.charAt(5);
+                if (Character.isDigit(temp)) {
+                    int item = Integer.parseInt(input.substring(5, 6));
+                    taskList.done(item - 1);
+                    storage.done(item - 1);
+                    ui.showDone(taskList.get(item - 1));
+                    return;
+                }
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new DukeException("OOPS!!! The target of finished duke.task cannot be empty.");
+            }
+        } else if (lowerCase.startsWith("delete")) {
+            try {
+                char temp = input.charAt(7);
+                if (Character.isDigit(temp)) {
+                    int item = Integer.parseInt(input.substring(7, 8));
+                    Task task = taskList.get(item - 1);
+                    taskList.delete(item - 1);
+                    storage.delete(item - 1);
+                    ui.showDelete(task, taskList.length());
+                    return;
+                }
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new DukeException("OOPS!!! The target of deleting duke.task cannot be empty.");
+            }
+        } else if (lowerCase.startsWith("find")) {
+            try{
+                String keyword = lowerCase.substring(5).trim();
+                findKeyword(keyword);
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new DukeException("OOPS!!! Keyword cannot be empty.");
+            }
+        } else {
+            throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    /**
+     * Fetches Task item from TaskList and print through Ui
+     */
+    public void listCommand() {
+        String items = "";
+        if (taskList.length() > 0) {
+            items = items + "1. " + taskList.get(0).toString();
+        }
+        for (int i = 2 ; i <= taskList.length(); i++) {
+            items = items + "\n    " + i + ". " + taskList.get(i-1).toString();
+        }
+        ui.showList(items);
+    }
+
+    /**
+     * Determines if the command is to exit the program.
+     *
+     * @param command the input
+     * @return true or false
+     */
+    public boolean isExit(String command) {
+        return command.toLowerCase().equals("bye");
+    }
+
+    public void findKeyword(String keyword) {
```

Comment:
Would it be good to have a JavaDoc for this method?

Reply:
> Would it be good to have a JavaDoc for this method?

Thanks for pointing out those non-standard coding style that I didn't realize!